src/components/CodeHighlighting.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <code class="code-highlighting" :class="`code-highlighting--${language ?? 'none'}`">
+    <CodeLines :code="code" :show-line-numbers="showLineNumbers" />
+  </code>
+</template>
+
+<script lang="ts" setup>
+  import { highlight } from 'prismjs'
+  import { computed } from 'vue'
+  import CodeLines from '@/components/CodeLines.vue'
+  import { definition as json } from '@/utilities/languageDefinitions/json'
+  import { definition as python } from '@/utilities/languageDefinitions/python'
+
+  const props = defineProps<{
+    value?: string,
+    language?: 'json' | 'python',
+    showLineNumbers?: boolean,
+  }>()
+
+  const grammar = computed(() => {
+    switch (props.language) {
+      case 'json':
+        return json
+      case 'python':
+        return python
+      default:
+        return undefined
+    }
+  })
+
+  const code = computed(() => {
+    if (props.language && grammar.value) {
+      return highlight(props.value ?? '', grammar.value, props.language)
+    }
+
+    return props.value ?? ''
+  })
+</script>
+
+<style>
+.code-highlighting {
+  -moz-tab-size: 4;
+  -o-tab-size: 4;
+  tab-size: 4;
+
+  -webkit-hyphens: none;
+  -moz-hyphens: none;
+  -ms-hyphens: none;
+  hyphens: none;
```

Comment:
Quick double check on if we actually need these vendor prefixes.

Reply:
I'll need @znicholasbrown to help with this one

Reply:
Lol https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens#browser_compatibility

Reply:
Seems like `hyphens` needs the prefixes but I don't think `tab-size` does according to mdn. 
https://developer.mozilla.org/en-US/docs/Web/CSS/tab-size
https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens

Reply:
the mdn hyphen support is nightmarish cause of localization but i think @pleek91 is right that tab size probably doesn't need it (also the vendor prefixes might be added by auto-prefixer regardless so it all might be moot)